src/common/commands_parser.rs

```diff
@@ -274,6 +280,137 @@ impl PcrArgs {
     }
 }
 
+/// The arguments used by `sign-eif` command
+#[derive(Debug, Clone)]
+pub struct SignArgs {
+    /// The method used for signing the EIF
+    pub signing_method: String,
+    /// The path to the enclave image file.
+    pub eif_path: String,
+    /// The path to the signing certificate.
+    pub signing_certificate: String,
+    /// The path to the private key.
+    pub private_key: Option<String>,
+    /// The region in which the KMS key resides.
+    pub region: Option<String>,
+    /// The KMS key id.
+    pub key_id: Option<String>,
+}
+
+impl SignArgs {
+    /// Construct a new `SignArg` instance from the given command-line arguments.
+    pub fn new_with(args: &ArgMatches) -> NitroCliResult<Self> {
+        let signing_method = parse_signing_method(args)
```

Comment:
This `signing_method` argument is really not required. It should be indicated by either the presence of `key_id` or `private_key` and no need to have a separate argument for it.